### PR TITLE
mtproxy: Add package

### PR DESCRIPTION
**Pull request by yangfl**
Signed-off-by: David Yang <mmyangfl@gmail.com>

### CODE REVIEW

**Comment by diizzyy** [comment]
What's the source for all patches? Upstream?

**Comment by yangfl** [comment]
written by me. The original source is a damn amd64 only, glibc only code.

**Comment by neheb** [comment]
@yangfl still want this in? Please rebase.

Looks like the entware people are also using this.

**Comment by yangfl** [comment]
@neheb done. Not sure if it's such useful since someone report it's CPU/memory hungry Entware/Entware#76

File: net/mtproxy/Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+PKG_NAME:=MTProxy
+PKG_VERSION:=1
+PKG_RELEASE:=1
+PKG_BUILD_PARALLEL:=1
+
+PKG_SOURCE_URL:=https://codeload.github.com/TelegramMessenger/MTProxy/tar.gz/v$(PKG_VERSION)?
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_HASH:=a1fc704519c84d9b3578c4063476775a54630a5fb007f745ac35ad8e4071d2ef
+PKG_LICENSE:=GPL-2+
```
Comment (neheb) [inline net/mtproxy/Makefile]: PKG_LICENSE_FILES should be added.

**Comment by yangfl** [comment]
@neheb ok

File: net/mtproxy/Makefile (hunk 1 above)
Comment (diizzyy) [reply net/mtproxy/Makefile]: Not a valid SPDX tag

Comment (yangfl) [reply net/mtproxy/Makefile]: Hmm what is suitable?

**Comment by neheb** [comment]
As far as CPU usage goes, this was linked in the thread: https://github.com/9seconds/mtg/

Since we have go in OpenWrt, could use that instead.

**Comment by ryzhovau** [comment]
@neheb golang written daemon? No, it will heavy too. 
Entware users [tested](https://t.me/Entware/1177) python variant. Looks like fine to embedded devices.

**Comment by neheb** [comment]
Does that variant work with Python 3?

**Comment by Kirya2k** [comment]
> Does that variant work with Python 3?
Yep

**Comment by yangfl** [comment]
There are some patches TelegramMessenger/MTProxy#202 trying to address them. However I'm so busy these days so any help/tests is welcomed.

**Comment by yangfl** [comment]
I will not go on with this package any more. This is the most awkward code I've ever seen. Just take a closer look and you'll know how they ignore every bit of good practice and peacock their worthless skill of micro-optimization. But for anyone who are interested, feel free to pick any code that might be useful.

**Comment by predators46** [comment]
@yangfl 

thanks

```
root@OpenWrt:~# opkg install mtproxy_1-1_aarch64_generic.ipk
Installing mtproxy (1-1) to root...
Configuring mtproxy.
Downloading 'https://core.telegram.org/getProxySecret'
Connecting to 149.154.167.99:443
Writing to '/var/lib/mtproto-proxy/proxy-secret'
/var/lib/mtproto-pro 100% |*******************************|   128   0:00:00 ETA
Download completed (128 bytes)
Downloading 'https://core.telegram.org/getProxyConfig'
Connecting to 149.154.167.99:443
Writing to '/var/lib/mtproto-proxy/proxy-multi.conf'
/var/lib/mtproto-pro 100% |*******************************|   752   0:00:00 ETA
Download completed (752 bytes)
/usr/bin/mtproto-proxy -u nobody -H 3907 -S CHANGEME -6 --aes-pwd /var/lib/mtproto-proxy/proxy-secret -M 1 --cpu-threads 2 --io-threads 4 /var/lib/mtproto-proxy/proxy-multi.conf
root@OpenWrt:~# mtproto-proxy --help
[27142][2023-12-28 12:42:03.357736 local] Invoking engine mtproxy-0.01 compiled at Nov
14 2023 13:38:11 by gcc 12.3.0 64-bit after commit 1
usage: mtproto-proxy [-v] [-6] [-p<port>] [-H<http-port>{,<http-port>}] [-M<workers>] [-u<username>] [-b<backlog>] [-c<max-conn>] [-l<log-name>] [-W<window-size>] <config-file>
mtproxy-0.01 compiled at Nov 14 2023 13:38:11 by gcc 12.3.0 64-bit after commit 1
        Simple MT-Proto proxy
        --ipv6/-6                               enables ipv6 TCP/UDP support
        --max-special-connections/-C <arg>      sets maximal number of accepted client
connections per worker
        --http-ports/-H <arg>                   comma-separated list of client (HTTP) ports to listen
        --slaves/-M <arg>                       spawn several slave workers
        --proxy-tag/-P <arg>                    16-byte proxy tag in hex mode to be passed along with all forwarded queries
        --random-padding-only/-R                allow only clients with random padding
option enabled
        --mtproto-secret/-S <arg>               16-byte secret in hex mode
        --ping-interval/-T <arg>                sets ping interval in second for local
TCP connections (default 5.000)
        --window-clamp/-W <arg>                 sets window clamp for client TCP connections
        --backlog/-b <arg>                      sets backlog size
        --connections/-c <arg>                  sets maximal connections number
        --daemonize/-d {arg}                    changes between daemonize/not daemonize mode
        --help/-h                               prints help and exits
        --log/-l <arg>                          sets log file name
        --port/-p <arg>                         <port> or <sport>:<eport> sets listening port number or port range
        --user/-u <arg>                         sets user name to make setuid
        --verbosity/-v {arg}                    sets or increases verbosity level
        --aes-pwd <arg>                         sets custom secret.conf file
        --nice <arg>                            sets niceness
        --msg-buffers-size <arg>                sets maximal buffers size (default 268435456)
        --disable-tcp                           do not open listening tcp socket
        --crc32c                                Try to use crc32c instead of crc32 in tcp rpc
        --cpu-threads <arg>                     Number of CPU threads (1-64, default 8)        --io-threads <arg>                      Number of I/O threads (1-64, default 16)
        --allow-skip-dh                         Allow skipping DH during RPC handshake
        --force-dh                              Force using DH for all outbound RPC connections
        --max-accept-rate <arg>                 max number of connections per second that is allowed to accept
        --max-dh-accept-rate <arg>              max number of DH connections per second that is allowed to accept
        --multithread {arg}                     run in multithread mode
        --tcp-cpu-threads <arg>                 number of tcp-cpu threads
        --tcp-iothreads <arg>                   number of tcp-io threads
        --nat-info <arg>                        <local-addr>:<global-addr>      sets network address translation for RPC protocol handshake
        --address <arg>                         tries to bind socket only to specified
address
        --http-stats                            allow http server to answer on stats queries
root@OpenWrt:~#
```

**Pull request closed without merging**